### PR TITLE
I can haz RGB 😄 

### PR DESCRIPTION
**Pull request by pkendall64**
Solder a WS2812B RGB LED to the DIY TX module.

This has been updated to be 4 better colors for the rates
| Rate 2.4Ghz | Rate 900Mhz | Color |
|---|---|---|
| 500/250hz | 200hz | Green |
| 250/150hz | 100hz | Yellow |
| 150/50hz | 50hz | Orange |
| 50/25hz | 25hz | Red |

The brightness of the LED when connected indicates the current power level of the TX module.

Also, theres a rainbow wave of color at startup for funzies 😈

### CODE REVIEW

**Comment by JyeSmith** [comment]
Woohoo, the LED is back 😄

**Review by JyeSmith — APPROVED** [review]

**Comment by XSJFYZS** [comment]
Cool, love it，this is my changed PCB, but it has not been tested. If there is a better design, please share it with me
[E28-2G4M27S + ESP-01F_RGB 2021-05-13_2021-05-13.zip](https://github.com/ExpressLRS/ExpressLRS/files/6472390/E28-2G4M27S.%2B.ESP-01F_RGB.2021-05-13_2021-05-13.zip)

File: src/src/LED.h [hunk 1]
```diff
@@ -21,18 +21,9 @@ HslColor hslBlack(black);
...
-    }
-    if (ExpressLRS_currAirRate_Modparams->enum_rate == RATE_50HZ)
-    {
-        strip.ClearTo(RgbColor(LED_MAX_BRIGHTNESS, 0, 0));
-    }
-  strip.Show();
+    #define ON(on) ((ExpressLRS_currAirRate_Modparams->enum_rate & on) ? 0 : LED_MAX_BRIGHTNESS)
+    strip.ClearTo(RgbColor(ON(0b100), ON(0b010), ON(0b001)));
+    strip.Show();
+    #undef ON
```
Comment (michidk) [inline src/src/LED.h:27]: I find this new version harder to read and understand (especially without comments) than the first one (though `else if` should be used). Is there a reason, that I am currently not aware of, for this rewrite, other than that it uses less LoC?

**Comment by michidk** [comment]
I had trouble understanding the color computation at first (without comments the macro is not very intuitive to read imho). 

But the coloring is now like this:
RATE_500HZ = 0 -> white (1, 1, 1)
RATE_250HZ = 1 -> yellow (1, 1, 0)
RATE_200HZ = 2 -> purple (1, 0, 1)
RATE_150HZ = 3 -> red (1, 0, 0)
RATE_100HZ = 4 -> cyan (0, 1, 1)
RATE_50HZ  = 5 -> green (0, 1, 0)
(with 1 being LED_MAX_BRIGHTNESS)

I am not sure the color would add any value, since I am not sure I could recall this table on the field.
Maybe we can come up with a more natural color scheme?

I was thinking something like this
RATE_500HZ = 0 -> red (1, 0, 0)
RATE_250HZ = 1 -> orange (255, 255/2, 0)
RATE_200HZ = 2 -> yellow (1, 1, 0)
RATE_150HZ = 3 -> white (1, 1, 1)
RATE_100HZ = 4 -> cyan (0, 1, 1)
RATE_50HZ  = 5 -> blue (0, 0, 1)
(with 1 being LED_MAX_BRIGHTNESS)

**Comment by caixiaohai0** [comment]
> 呜呜，LED又回来了
PIN
Can we change the LED of pin27 to sw2812?

**Comment by pkendall64** [comment]
Yes I think that will work if you are using this branch for that target.

**Comment by alexdibi** [comment]
ledpcb
wanting to do it on the slim pcb I should connect the led in this way, right?
the +5 input of the led on the 3.3v output of the regulator ...

**Comment by caixiaohai0** [comment]
> ledpcb
> wanting to do it on the slim pcb I should connect the led in this way, right?
> the +5 input of the led on the 3.3v output of the regulator ...

NO,WS2812 VCC=3.5~5.5V,You have to be in this voltage range

**Comment by alexdibi** [comment]
> > ledpcb
> > wanting to do it on the slim pcb I should connect the led in this way, right?
> > the +5 input of the led on the 3.3v output of the regulator ...
> 
> NO,WS2812 VCC=3.5~5.5V,You have to be in this voltage range

well with the regulator set as the instructions we are on 3.47v, is the PIN of the signal right anyway?

**Comment by caixiaohai0** [comment]
> > > ledpcb
> > > wanting to do it on the slim pcb I should connect the led in this way, right?
> > > the +5 input of the led on the 3.3v output of the regulator ...
> > 
> > 
> > NO,WS2812 VCC=3.5~5.5V,You have to be in this voltage range
> 
> well with the regulator set as the instructions we are on 3.47v, is the PIN of the signal right anyway? 

right

**Comment by pkendall64** [comment]
I measured 3.47 volts on mine after adjusting the buck converter as in the diagram and I have performed the surgery of 5 DIY 2400 units and they all work. Also, the LED works when the TX is plugged in via USB as well which is only providing 3.3V to the led, so they may work with at 3.3V.

**Review by cruwaller — APPROVED** [review]

**Review by michidk — APPROVED** [review]

**Comment by ariefadha** [comment]
i have read on arduino forum that logic high input will be detected at a minimum voltage of 0.7 * Vcc
and ws2812 cant run on 3v3 input. It might difference manufacture has problem with 3v3 so i have several ws2812, ill check if they are good with 3v3 and let you know here

**Comment by cruwaller** [comment]
I have run ws2812 leds from 3V3 source and works correctly.

**Comment by JyeSmith** [comment]
Changing rf mode while connected does not change the LED colour.

**Comment by pkendall64** [comment]
You mean the rate? I just tried that and it works for me! Also, there are different RGB LEDS, it's currently set to use a GRB LED, but if you comment out the targets.h line 530 it will work for RGB LEDs. I think it needs a define in user_defines.txt for the type of LED.

**Comment by alexdibi** [comment]
since we're at it ... wouldn't it be useful to have a different brightness or pulsation for each power level used?

**Comment by pkendall64** [comment]
I currently use dimmed pulsation to indicate that it is not connected to a receiver. I'll look into dimming the solid color when connected for the power level.

**Comment by pkendall64** [comment]
Dues to popular demand the LED brightness is adjusted based on selected power level when connected to a receiver.

**Pull request merged**